### ISSUE TITLE
Datalake_store - Firewall Rule, name parameter is required

### ISSUE THREAD

**Issue by veronicagg** [Azure/azure-rest-api-specs#1673]
Customer report: https://github.com/Azure/azure-sdk-for-ruby/issues/900

azure_mgmt_datalake_store - Firewall Rule object, the name parameter is marked as an optional parameter here but it is required.
Please make it required in the documentation.

I believe the name parameter comes from https://github.com/Azure/azure-rest-api-specs/blob/current/specification/datalake-store/resource-manager/Microsoft.DataLakeStore/2016-11-01/account.json#L1573, which is not marked as required. 
Please evaluate and consider updating it, so the SDKs can expose the property this way.

**Comment by veronicagg** [comment]
@ro-joowan could you evaluate this issue?

**Comment by ro-joowan** [comment]
@veronicagg may I ask which API call is the customer invoking?  Is it this API (https://github.com/Azure/azure-rest-api-specs/blob/current/specification/datalake-store/resource-manager/Microsoft.DataLakeStore/2016-11-01/account.json#L45)?

**Comment by veronicagg** [comment]
@muhammad-asad-confiz may give you more details, as author of the original issue (Azure/azure-sdk-for-ruby#900). thanks!

**Comment by ro-joowan** [comment]
@muhammad-asad-confiz the more information you can provide the better -- which API was invoked and how the parameters were set.  Thank you!

**Comment by muhammad-asad-confiz** [comment]
@ro-joowan I am using azure-sdk-for-ruby and the sdk itself is making the API call.

**Comment by ro-joowan** [comment]
@muhammad-asad-confiz Sorry, I meant which method were you using and what were the parameters?

**Comment by muhammad-asad-confiz** [comment]
```
#<Azure::ARM::DataLakeStore::Models::DataLakeStoreAccount:0x0055ab093daa18 
@name="fogtestdls", 
@location="East US 2", 
@type="Microsoft.DataLakeStore/accounts", 
@firewall_state="Enabled", 
@firewall_allow_azure_ips="Enabled", 
@firewall_rules=[
#<Azure::ARM::DataLakeStore::Models::FirewallRule:0x0055ab09f92950 
@name="one", 
@start_ip_address="116.58.62.100", 
@end_ip_address="116.58.62.110">
], 
@encryption_state="Enabled", 
@encryption_config=
#<Azure::ARM::DataLakeStore::Models::EncryptionConfig:0x0055ab09f86bf0 
@type="UserManaged", 
@key_vault_meta_info=
#<Azure::ARM::DataLakeStore::Models::KeyVaultMetaInfo:0x0055ab09f7e4f0 @key_vault_resource_id="/subscriptions/My Subscription ID/resourceGroups/TestRG-ZN/providers/Microsoft.KeyVault/vaults/testdlskeyvault", 
@encryption_key_name="testdlskey", 
@encryption_key_version='My Key Version'
>>, 
@new_tier="Consumption", 
@current_tier="Consumption"
>
```

This is object which is created and specifies all parameters and this object is passed to create method of SDK 
https://github.com/Azure/azure-sdk-for-ruby/blob/8d9ac49c327561f6e6ab5454ba4911c81c0e0fa4/management/azure_mgmt_datalake_store/lib/generated/azure_mgmt_datalake_store/account.rb#L37

**Comment by ro-joowan** [comment]
@muhammad-asad-confiz Thank you for that.  What is the error message that you get when don't provide the FirewallRule name?

@matt1883 I confirmed with Rolland that the FirewallRule name is required whether you are just creating the FirewallRule or creating it during an ADLS account creation.  Should I make this required in the FirewallRule -> SubResource object? (https://github.com/Azure/azure-rest-api-specs/blob/current/specification/datalake-store/resource-manager/Microsoft.DataLakeStore/2016-11-01/account.json#L1573)

**Comment by ro-joowan** [comment]
@muhammad-asad-confiz after looking into this further, I agree with you that this needs to be required.  I will make this fix in the next update and will link my PR here when I open it.

Thank you!!

**Comment by ro-joowan** [comment]
@muhammad-asad-confiz, @veronicagg I am addressing this bug in this PR: https://github.com/Azure/azure-rest-api-specs/pull/1999

**Comment by veronicagg** [comment]
@ro-joowan thanks, be aware that some changes may be considered breaking for the SDK and could impact CLI/Powershell commands, since it appears that modifications are in existing api-versions. @jianghaolu is the assigned reviewer of the PR - tagging him here, so he's aware.

**Comment by ro-joowan** [comment]
@veronicagg, Yes, technically this would be a breaking change.  But the good news is that the service required these parameters (e.g., FirewallRule name addressed in this Issue) in the first place.  Without providing it, an error would be thrown by the service as @muhammad-asad-confiz ran here.  Thank you for noting this!

**Comment by veronicagg** [comment]
@ro-joowan agree.

**Comment by ro-joowan** [comment]
@veronicagg @muhammad-asad-confiz, This has been addressed in this merged PR: https://github.com/Azure/azure-rest-api-specs/pull/1999/

Thanks for bringing it up!

**Comment by ro-joowan** [comment]
@veronicagg Please close :)

**Comment by lmazuel** [comment]
Double-checked with @ro-joowan as fixed indeed in #1999

**Comment by ro-joowan** [comment]
FYI @matt1883 -- in the ADLA/ADLS Account object refactor PR, the Firewall rule name parameter during Account creation is now required.  It's so clean - I love it!
